scr: release v3.1.0, including components

CamStan opened the pull request:
SCR and the SCR components have new releases

- AXL v0.9.0
- ER v0.5.0
- KVTREE v1.5.0
- Rankstr v0.4.0
- Shuffile v0.4.0
- Spatha v0.4.0
- dtcmp v1.1.5
- lwgrp v1.0.6
- Redset v0.4.0
  - New variants added to Redset

- SCR v3.1.0
  - Added Flux resource manager
  - Added pthreads variant
  - Removed deprecated release candidates and references
  - Cleaned up component dependency versions
  - Updated versions within variants and cleaned up cmake_args

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

alalazo reviewed (changes requested):
There are pipeline failures, for instance: https://gitlab.spack.io/spack/spack/-/jobs/12236018

CamStan committed:
scr: release v3.1.0, including components

SCR and the SCR components have new releases

- AXL v0.9.0
  - MPI variant added to AXL package
- ER v0.5.0
- KVTREE v1.5.0
- Rankstr v0.4.0
- Shuffile v0.4.0
- Spatha v0.4.0
- dtcmp v1.1.5
- lwgrp v1.0.6
- Redset v0.4.0
  - New variants added to Redset

- SCR v3.1.0
  - Added Flux resourse manager
  - Added pthreads variant
  - Removed deprecated release candidates and references
  - Cleaned up component dependency versions
  - Updated versions within variants and cleaned up cmake_args

CamStan commented:
> There are pipeline failures, for instance: https://gitlab.spack.io/spack/spack/-/jobs/12236018

@alalazo apparently I did push this fix. This should be ready now.

alalazo commented on var/spack/repos/builtin/packages/redset/package.py:
```diff
@@ -38,7 +39,11 @@ class Redset(CMakePackage):
     depends_on("rankstr@0.3.0:", when="@0.3.0:")
 
-    variant("shared", default=True, description="Build with shared libraries")
+    variant("cuda", default=False, description="Enable CUDA support", when="@0.4:")
```
Why not making this a `CudaPackage`?

CamStan replied:
Totally missed that, thank you.

alalazo commented on var/spack/repos/builtin/packages/scr/package.py:
```diff
@@ -79,39 +72,26 @@ class Scr(CMakePackage):
-    depends_on("shuffile@0.2.0", when="@3.0.1")
-    depends_on("spath@0.2.0 +mpi", when="@3.0.1")
-    depends_on("dtcmp@1.1.4", when="@3.0.1")
+    depends_on("axl@0.8.0: +mpi", when="@3.1.0")
+    depends_on("er@0.5.0", when="@3.1.0")
+    depends_on("kvtree@1.4.0:", when="@3.1.0")
+    depends_on("rankstr@0.3.0:", when="@3.1.0")
+    depends_on("redset@0.4.0", when="@3.1.0")
+    depends_on("shuffile@0.3.0:", when="@3.1.0")
+    depends_on("spath@0.3.0: +mpi", when="@3.1.0")
+    depends_on("dtcmp@1.1.5", when="@3.1.0")
```
Here you can group for better readability:
```suggestion
    with when("@3.1.0"):
        depends_on("axl@0.8.0: +mpi")
        depends_on("er@0.5.0")
        depends_on("kvtree@1.4.0:")
        depends_on("rankstr@0.3.0:")
        depends_on("redset@0.4.0")
        depends_on("shuffile@0.3.0:")
        depends_on("spath@0.3.0: +mpi")
        depends_on("dtcmp@1.1.5")
```
and similarly below

CamStan replied:
Updated to this format.

alalazo reviewed (commented):
Essentially LGTM, just a couple of comments to improve readability of the recipe.
